src/ServiceProviders/RepositoryProviders/GitHub/GitHubProvider.php

```diff
@@ -210,11 +210,11 @@ function ($item) {
             },
             $data
         ), 1, 0);
- 
+
         return $branchList;
      }
 
-    protected function gitHubAPI($uri, $data = [], $method = 'GET')
+    protected function gitHubAPI($uri, $data = [], $method = 'GET', $follow_next_link = FALSE)
```

Comment:
This will take the cautious approach and only enable this behavior if explicitly requested. However I think this precaution may be lifted with the current code. The only limitation would be that we would no longer be capable of requesting a single page. I think requesting a single page is a useless feature, as pages are merely an implementation detail to mitigate unforeseen performance implications. Such performance degradation is preferred in a CLI application in favor of correctness.

@greg-1-anderson please let me know if you'd like me to lift this guard.

Reply:
I'm -1 on the extra parameter. I'm undecided at this point if the feature needs to be guarded.

Sometimes you might make an API call just to get the first response. Automatically traversing all pages would not be desirable in that instance, as some paged responses, e.g. a search, could have many many pages.

Reply:
I'll look at this more later with an opinion on whether there should be a parameter here, or a separate method for paged vs unpaged access.